src/docmanager/index.ts

```diff
@@ -256,66 +230,56 @@ class DocumentManager {
   /**
    * Register a widget factory with the document manager.
    *
-   * @params options - The options used to register the factory.
+   * @params factory - An instance of a widget factory.
    *
    * @returns A disposable used to unregister the factory.
    */
-  registerWidgetFactory(options: IWidgetFactoryOptions): IDisposable {
+  registerWidgetFactory(factory: IWidgetFactory): IDisposable {
     return void 0;
   }
 
   /**
    * Register the default widget factory of a given file type.
    *
-   * @params options - The options used to register the factory.
+   * @params factory - An instance of a widget factory.
    *
    * @returns A disposable used to unregister the factory.
    */
-  registerDefaultWidgetFactory(options: IWidgetFactoryOptions): IDisposable {
+  registerDefaultWidgetFactory(factory: IWidgetFactory): IDisposable {
     return void 0;
   }
 
   /**
    * Register a model factory.
    *
-   * @param options - The options used to register the factory.
+   * @param factory - An instance of a model factory.
    *
    * @returns A disposable used to unregister the factory.
    */
-  registerModelFactory(options: IModelFactoryOptions): IDisposable {
+  registerModelFactory(factory: IModelFactory): IDisposable {
     return void 0;
   }
 
   /**
-   * Register a file type.
-   *
-   * @param fileType - the name of the file type.
+   * Register a file type and associated extensions.
    *
-   * @param extensions - a list of file extensions to associate with the type.
+   * @param name - The name of the file type.
    *
-   * @param canCreate - whether a file type can be used by [createNew].
+   * @param extensions - The list of file extensions associated with the type.
    *
-   * @returns A disposable which can be used to unregister the file type.
-   *
-   * #### Notes
-   * The default fileType is 'File', will be used when no matching extensions
-   * are found and can be used to create new files.
-   */
-  registerFileType(fileType: string, extensions: string[], canCreate=false): IDisposable {
-    this._fileTypes[fileType] = {
-      extensions,
-      canCreate
-    }
-    return void 0;
+   * @returns A disposable used to unregister the file type.
+   */
+  registerFileType(name: string, extensions: string[]): IDisposable {
```

Comment:
I like your idea below, that the extensions argument in registering defaults is optional.